### PR TITLE
Let @[equational_result] handle facts

### PR DESCRIPTION
**Pull request by nomeata**

### CODE REVIEW

File: equational_theories/ParseImplications.lean [hunk 1]
```diff
@@ -55,15 +64,28 @@ def parseNonimplication (thm_ty : Expr) : MetaM (Option Implication) := do
...
-            | _ => return none
-          | _ => return none
+          let some facts := go #[ty1] #[] #[] | return none
+          if facts.satisfied.isEmpty && facts.refuted.isEmpty then return none
+          return .some facts
       | _ => return none
   | _ => return none
+where
+  -- NB: This is written as a tail-recursive function, else some large facts statements
+  -- cauase this to blow the stack
```
Comment (dwrensha) [inline equational_theories/ParseImplications.lean:74]: cauase -> cause

File: equational_theories/ParseImplications.lean [hunk 2]
```diff
@@ -12,6 +12,15 @@ structure Implication where
   rhs : String
 deriving Lean.ToJson, Lean.FromJson
 
+/--
+Names of equations satisfied rsp. refuted by the same magma; used to concisely express
+antiimplications.
```
Comment (dwrensha) [inline equational_theories/ParseImplications.lean:17]: Either here or in `FactsSyntax.lean` (probably there is better) I'd like to see it explicitly worked out which negated implications a `Facts` entails. (I, for one, have not internalized De Morgan's laws deeply enough for this to be immediately obvious to me.)

For example, if I understand correctly,
```
Facts G [1, 2] [4, 5]
```
entails `not (Equation4 -> Equation1)` and `not (Equation4 -> Equation2)` and `not (Equation5 -> Equation1)` and `not (Equation5 -> Equation2).`

**Review by dwrensha — COMMENTED** [review]
Looks good to me!

File: equational_theories/ParseImplications.lean (hunk 1 above)
Comment (nomeata) [reply equational_theories/ParseImplications.lean:74]:
```suggestion
  -- cause this to blow the stack
```

File: equational_theories/ParseImplications.lean (hunk 2 above)
Comment (nomeata) [reply equational_theories/ParseImplications.lean:17]: Almost: It proofs that these implications don’t hold: 1→4, 1→5, 2→4, 2→5. The first list is the list of equations that are satisfied.

Comment (nomeata) [reply equational_theories/ParseImplications.lean:17]:
```suggestion
antiimplications.

For example, if `satisfied = [1, 2]` and `refuted = [4, 5]`, then this expresses
the four antiimplications `¬ 1→4`, `¬ 1→5`, `¬ 2→4`, `¬ 2→5`.
```

Comment (dwrensha) [reply equational_theories/ParseImplications.lean:17]: Thanks. My current code on `main` gets this backwards! (Though maybe any downstream consumers re-flip it.)

File: equational_theories/EquationalResult.lean [hunk 3]
```diff
@@ -97,25 +97,25 @@ elab_rules : command
...
     match res with
     | .implication ⟨lhs, rhs⟩ => println! "{name}: {lhs} → {rhs}"
-    | .nonimplication ⟨lhs, rhs⟩ => println! "{name}: ¬ ({lhs} → {rhs})"
+    | .facts ⟨satisfied, refuted⟩ => println! "{name}: {satisfied} // {refuted}"
     | .unconditional rhs => println! "{name}: {rhs} holds unconditionally"
 
 --- Output of the extract_implications executable.
 structure Output where
   implications : Array Implication
-  nonimplications : Array Implication
```
[the remark this answers is not in the document]
Comment (nomeata) [reply equational_theories/EquationalResult.lean:106]: We have a widget? Sorry, didn’t know that (and CI didn’t tell me). I’ll see if I can fix it.

[the remark this answers is not in the document]
Comment (dwrensha) [reply equational_theories/EquationalResult.lean:106]: `Visualization.lean`.

[the remark this answers is not in the document]
Comment (nomeata) [reply equational_theories/EquationalResult.lean:106]: Outlined a fix in https://github.com/teorth/equational_theories/pull/138

**Pull request merged**